app/code/Magento/Config/Block/System/Config/Form.php

```diff
@@ -366,9 +366,12 @@ protected function _initElement(
 
         $sharedClass = $this->_getSharedCssClass($field);
         $requiresClass = $this->_getRequiresCssClass($field, $fieldPrefix);
+        $isReadOnly = $this->isDefaultFieldReadOnly($path);
```

Comment:
It would be nice to move all the readonly check to a private function

Reply:
I have fixed this, moved all checks in private function